test/test_shadowState.py

```diff
@@ -0,0 +1,39 @@
+from unittest import TestCase
```

Comment:
trivial: do lowercase file name like `test_shadow_state.py` or `test_shadowstate.py`

the wider world of python style isn't 100% consistent, but the style we used, because it seemed most common, is: class names are capitalized but everything else (variables, filenames, etc) are lowercase